### PR TITLE
Misc changes from private repo

### PR DESCRIPTION
**Pull request by JCash**
## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [ ] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

----------

Example of a well written PR description:

1. Start with the user facing changes. This will end up in the release notes.
1. Add one of the GitHub approved closing keywords
1. Optionally also add the technical changes made. This is information that might help the reviewer. It will not show up in the release notes. Technical changes are identified by a line starting with one of these:
   1. `### Technical changes` 
   1. `Technical changes:`
   2. `Technical notes:`

```
There was a anomaly in the carbon chroniton propeller, introduced in version 8.10.2. This fix will make sure to reset the phaser collector on application startup.

Fixes #1234

### Technical changes
* Pay special attention to line 23 of phaser_collector.clj as it contains some interesting optimizations
* The propeller code was not taking into account a negative phase.
```

### CODE REVIEW

File: ci/ci.py [hunk 1]
```diff
@@ -479,7 +479,7 @@ def main(argv):
                 return
 
         if platform and not is_platform_private(platform):
-            if platform not in ['x86_64-win32']:
+            if platform not in ['x86_64-win32', 'x86_64-linux']:
```
Comment (JCash) [inline ci/ci.py:482]: For our defoldsdk.zip to build with linux tools (dlib_shared.so), we also need to build Linux

File: engine/dlib/src/dlib/ssdp.cpp [hunk 2]
```diff
@@ -769,7 +769,7 @@ namespace dmSSDP
         if (sr != dmSocket::RESULT_OK)
         {
             // When returning from sleep mode on iOS socket is in state ECONNABORTED
-            if (sr == dmSocket::RESULT_CONNABORTED || sr == dmSocket::RESULT_NOTCONN)
+            if (sr == dmSocket::RESULT_CONNABORTED || sr == dmSocket::RESULT_NOTCONN || sr == dmSocket::RESULT_NETDOWN)
```
Comment (JCash) [inline engine/dlib/src/dlib/ssdp.cpp:772]: An issue in debug mode where, if the app comes back after sleep, the connection is lost, but we just continue the loop checking for more content (-> hang)

File: engine/gameobject/src/gameobject/test/delete/test_gameobject_delete.cpp [hunk 3]
```diff
@@ -16,6 +16,7 @@
 
 #include <algorithm>
 #include <map>
+#include <random>
```
Comment (JCash) [inline engine/gameobject/src/gameobject/test/delete/test_gameobject_delete.cpp:19]: C++17 hsa removed some old functionality.

File: engine/gamesys/src/gamesys/test/test_gamesys.cpp [hunk 4]
```diff
@@ -4524,6 +4524,8 @@ TEST_F(RenderConstantsTest, CreateDestroy)
     dmGameSystem::DestroyRenderConstants(constants);
 }
 
+#if !defined(DM_PLATFORM_VENDOR) // we need to fix our test material/shader compiler to work with the constants
```
Comment (JCash) [inline engine/gamesys/src/gamesys/test/test_gamesys.cpp:4527]: Some tests that don't work currently for the private platforms

File: engine/graphics/src/opengl/graphics_opengl.cpp [hunk 5]
```diff
@@ -1526,11 +1526,12 @@ static void LogFrameBufferError(GLenum status)
 #endif
     }
 
-    static void OpenGLFlip(HContext context)
+    static void OpenGLFlip(HContext _context)
     {
         DM_PROFILE(__FUNCTION__);
-        PostDeleteTextures((OpenGLContext*) context, false);
-        dmPlatform::SwapBuffers(((OpenGLContext*) context)->m_Window);
+        OpenGLContext* context = (OpenGLContext*) _context;
```
Comment (JCash) [inline engine/graphics/src/opengl/graphics_opengl.cpp:1532]: Cosmetic fix, easier to hover over in debugger.

File: engine/hid/src/test/test_app_hid.cpp [hunk 6]
```diff
@@ -191,6 +191,7 @@ static void* EngineCreate(int argc, char** argv)
 
 static const char* KeyToStr(dmHID::Key key)
 {
+#if !defined(DM_PLATFORM_VENDOR)
```
Comment (JCash) [inline engine/hid/src/test/test_app_hid.cpp:194]: It was commented out on the private platform.
Better to use defines to control code.

File: scripts/build.py [hunk 7]
```diff
@@ -47,8 +47,12 @@
...
     sys.modules['build_private'] = build_vendor
     print("Imported %s from %s" % ('build_private', build_vendor.__file__))
-except ModuleNotFoundError:
-    pass
+except ModuleNotFoundError as e:
+    if "No module named 'build_vendor'" in str(e):
+        print("Couldn't find build_vendor.py. Skipping.")
+        pass
+    else:
+        raise e
```
Comment (JCash) [inline scripts/build.py:55]: Better error reporting when something goes awry

File: scripts/build.py [hunk 8]
```diff
@@ -1436,11 +1440,16 @@ def build_sdk(self):
...
+
+        # For the linux build tools (protoc, dlib_shared etc)
+        if 'x86_64-linux' not in platforms:
+            platforms.append('x86_64-linux')
+
+        # Since we usually want to use the scripts in this package on a linux machine, we'll unpack
+        # it last, in order to preserve unix line endings in the files
+        if 'x86_64-linux' in platforms:
+            platforms.remove('x86_64-linux')
+            platforms.append('x86_64-linux')
```
Comment (JCash) [inline scripts/build.py:1452]: It's good to remember that the different headers aren't always the same.
So, we choose linux as the common platform for headers.
It's still possible to have platform specific versions if needed.

File: scripts/copy_from_private_repo.py [hunk 9]
```diff
@@ -39,6 +39,8 @@
 FILE_PATTERNS.append('meta.properties') # TODO: create meta.edn plugins for extensions
 FILE_PATTERNS.append('build.xml') # TODO: make a strategy here
 
+FILE_PATTERNS.append('platform.sdks.json')
```
Comment (JCash) [inline scripts/copy_from_private_repo.py:42]: Probably not strictly necessary, and will always likely cause a conflict when merging from public to private, as the file will change version numbers a lot.

File: scripts/unpack_ddf.py [hunk 10]
```diff
@@ -23,6 +23,13 @@
 
 import os, sys
 
+try:
+    from google.protobuf import text_format
+except:
+    dynamo_home = os.environ.get('DYNAMO_HOME')
+    sys.path.append(os.path.join(dynamo_home, "lib", "python"))
+    sys.path.append(os.path.join(dynamo_home, "ext", "lib", "python"))
```
Comment (JCash) [inline scripts/unpack_ddf.py:31]: Make the unpack script easier to use, only requiring you to have DYNAMO_HOME set.

**Review by Jhonnyg — APPROVED** [review]

**Pull request merged**